### PR TITLE
feat: [PL-32183]: ListView for Roles

### PR DESCRIPTION
**Pull request by abhinavrastogi-harness**
### Summary

- Adds a new list view for roles listing page
- Adds role assignment counts to card view as well

#### Screenshots

Screenshot 2023-09-06 at 3 38 26 PM
Screenshot 2023-09-06 at 3 38 36 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/20-rbac/pages/Roles/views/RolesList.tsx [hunk 1]
```diff
@@ -140,6 +212,75 @@ const RolesList: React.FC = () => {
...
+          editRoleModal={editRoleModal}
+          openDeleteModal={openDeleteDialog}
+        />
+      </Popover>
+    )
+  }
+  const columns: Column<RoleResponse>[] = useMemo(
+    () => [
+      {
+        Header: 'Roles',
```
Comment (tan-nhu) [inline src/modules/20-rbac/pages/Roles/views/RolesList.tsx:254]: `getString()` for header texts

File: src/modules/20-rbac/components/RoleCard/RoleCard.tsx [hunk 2]
```diff
@@ -6,85 +6,31 @@
...
-import { useStrings } from 'framework/strings'
 import { getRoleIcon } from '@rbac/utils/utils'
 import type { ModulePathParams, ProjectPathProps } from '@common/interfaces/RouteInterfaces'
-import { ResourceType } from '@rbac/interfaces/ResourceType'
-import { PermissionIdentifier } from '@rbac/interfaces/PermissionIdentifier'
-import RbacMenuItem from '@rbac/components/MenuItem/MenuItem'
-import OpenInNewTab from '@rbac/components/MenuItem/OpenInNewTab'
+import useDeleteRoleDialog from '@rbac/hooks/useDeleteRoleDialog'
+import AssignedCounts from '@rbac/pages/Roles/views/AssignedCounts'
+import RoleMenu from '../RoleMenu/RoleMenu'
```
Comment (sunnykesh-harness) [inline src/modules/20-rbac/components/RoleCard/RoleCard.tsx:19]: Relative imports. Import using complete path '@rbac/...' will be helpful in any future refactoring.

**Review by sunnykesh-harness — APPROVED** [review]

File: src/modules/20-rbac/hooks/useDeleteRoleDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    cancelButtonText: getString('cancel'),
+    intent: Intent.DANGER,
+    buttonIntent: Intent.DANGER,
+    onCloseDialog: async (isConfirmed: boolean) => {
+      /* istanbul ignore else */ if (isConfirmed) {
+        try {
+          const deleted = await deleteRole(role.identifier, {
+            headers: { 'content-type': 'application/json' }
+          })
+          /* istanbul ignore else */ if (deleted) {
```
Comment (ishantmehtaharness) [inline src/modules/20-rbac/hooks/useDeleteRoleDialog.tsx:39]: else is there, do we need `ignore else`?

File: src/modules/20-rbac/pages/Roles/views/RolesList.tsx [hunk 4]
```diff
@@ -184,22 +327,45 @@ const RolesList: React.FC = () => {
...
+              )}
+            </div>
+            <Container className={css.pagination}>
+              <Pagination {...paginationProps} />
+            </Container>
+          </>
+        ) : (
+          <TableV2
+            columns={columns}
+            data={data?.data?.content?.filter(roleResponse => !isAccountBasicRole(roleResponse.role.identifier)) || []}
```
Comment (ishantmehtaharness) [inline src/modules/20-rbac/pages/Roles/views/RolesList.tsx:353]: Shall we memoize this data filtering, as it will create a new array every time?

**Review by ishantmehtaharness — APPROVED** [review]
2 nits, rest looks good

File: src/modules/20-rbac/pages/Roles/views/RolesList.tsx (hunk 1 above)
Comment (abhinavrastogi-harness) [reply src/modules/20-rbac/pages/Roles/views/RolesList.tsx]: fixed

File: src/modules/20-rbac/components/RoleCard/RoleCard.tsx (hunk 2 above)
Comment (abhinavrastogi-harness) [reply src/modules/20-rbac/components/RoleCard/RoleCard.tsx]: fixed

File: src/modules/20-rbac/pages/Roles/views/RolesList.tsx (hunk 4 above)
Comment (abhinavrastogi-harness) [reply src/modules/20-rbac/pages/Roles/views/RolesList.tsx]: done

File: src/modules/20-rbac/hooks/useDeleteRoleDialog.tsx (hunk 3 above)
Comment (abhinavrastogi-harness) [reply src/modules/20-rbac/hooks/useDeleteRoleDialog.tsx:39]: the else is not covered in the tests. it just shows a toast, so there's no test for it.

**Review by tan-nhu — APPROVED** [review]

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good

**Pull request merged**